oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3DSWithSSEKMS.java

```diff
@@ -0,0 +1,76 @@
+package org.apache.jackrabbit.oak.blob.cloud.s3;
+
+
+import java.io.ByteArrayInputStream;
+
+import org.apache.jackrabbit.core.data.DataRecord;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.jackrabbit.oak.blob.cloud.s3.S3DataStoreUtils.getS3Config;
+import static org.junit.Assert.fail;
+
+/**
+ * Test S3DataStore operation with SSE_S3 encryption.
+ * It requires to pass aws config file via system property  or system properties by prefixing with 'ds.'.
+ * See details @ {@link S3DataStoreUtils}.
+ * For e.g. -Dconfig=/opt/cq/aws.properties. Sample aws properties located at
+ * src/test/resources/aws.properties
+
+ */
+public class TestS3DSWithSSEKMS extends TestS3Ds {
+
+        protected static final Logger LOG = LoggerFactory.getLogger(TestS3DSWithSSES3.class);
+
+        @Override
+        @Before
+        public void setUp() throws Exception {
+            super.setUp();
+            props = getS3Config();
+            props.setProperty(S3Constants.S3_ENCRYPTION, S3Constants.S3_ENCRYPTION_SSE_KMS);
+            props.setProperty("cacheSize", "0");
+        }
+
+        /**
+         * Test data migration enabling SSE_KMS encryption.
+         */
+        @Test
+        public void testDataMigration() {
```

Comment:
rather than duplicating the whole method which need to be changed at all places, why not just add the test to the super class which calls an overridden method in he subclass altering behaviour/data as needed.